From d4ff521d7d40fd984215353a1c6097729a867308 Mon Sep 17 00:00:00 2001
From: Kjell Ahlstedt <kjellahlstedt@gmail.com>
Date: Mon, 3 Jul 2017 13:32:54 +0200
Subject: [PATCH] Gio: Exclude DesktopAppInfo from wrap_init.cc on MacOS

Modify the conditional inclusion of desktopappinfo.h in giomm.h and
wrap_init.cc. It shall not be included when running under MacOS. This should
have been fixed by commit dd671f29ff9a005776ebd9e7cf20fdc5e44c14b8.
Bug 781947
---
 configure.ac              | 6 ++++++
 gio/giomm.h               | 2 +-
 gio/src/desktopappinfo.hg | 4 +++-
 glib/glibmmconfig.h.in    | 3 +++
 4 files changed, 13 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 744e50b1..85fb4711 100644
--- configure.ac
+++ configure.ac
@@ -55,6 +55,8 @@ AS_CASE([$host_os], [mingw*], [glibmm_host_windows=yes], [glibmm_host_windows=no
 AC_MSG_RESULT([$glibmm_host_windows])
 AM_CONDITIONAL([HOST_WINDOWS_NATIVE], [test "x$glibmm_host_windows" = xyes])

+# This test for Mac OS is copied from glib. If the result of glib's test
+# is ever made available outside glib, use glib's result instead of this test.
 glib_have_cocoa=no
 AC_MSG_CHECKING([for Mac OS X Cocoa support])
 AC_TRY_CPP([
@@ -66,6 +68,10 @@ AC_TRY_CPP([

 AC_MSG_RESULT([$glib_have_cocoa])
 AM_CONDITIONAL(OS_COCOA, [test "$glib_have_cocoa" = "yes"])
+AS_IF([test "$glib_have_cocoa" = "yes"],
+[
+  AC_DEFINE([GLIBMM_OS_COCOA], [1], [Define only on Mac OS, COCOA])
+])

 # TODO: This makes no sense. --danielk
 AS_IF([test "x$enable_static" = xyes],
diff --git a/gio/giomm.h b/gio/giomm.h
index 3b79ed1e..d39df608 100644
--- gio/giomm.h
+++ gio/giomm.h
@@ -61,7 +61,7 @@
 #include <giomm/dbussubtreevtable.h>
 #include <giomm/dbusutils.h>
 #include <giomm/dbuswatchname.h>
-#ifndef G_OS_WIN32
+#if !defined(G_OS_WIN32) && !defined(GLIBMM_OS_COCOA)
 #include <giomm/desktopappinfo.h>
 #endif
 #include <giomm/drive.h>
diff --git a/gio/src/desktopappinfo.hg b/gio/src/desktopappinfo.hg
index ba654289..6a9f1305 100644
--- gio/src/desktopappinfo.hg
+++ gio/src/desktopappinfo.hg
@@ -23,6 +23,9 @@ _CONFIGINCLUDE(giommconfig.h)
 _DEFS(giomm,gio)
 _PINCLUDE(glibmm/private/object_p.h)

+// This file is not used on Windows and Mac OS Cocoa
+_GMMPROC_WRAP_CONDITIONALLY(if !defined(G_OS_WIN32) && !defined(GLIBMM_OS_COCOA))
+
 namespace Glib
 {

@@ -44,7 +47,6 @@ class DesktopAppInfo
 {
   _CLASS_GOBJECT(DesktopAppInfo, GDesktopAppInfo, G_DESKTOP_APP_INFO, Glib::Object, GObject)
   _IMPLEMENTS_INTERFACE(AppInfo)
-  _GTKMMPROC_WIN32_NO_WRAP

 public:
   // TODO: should use _WRAP_CREATE(), but these functions do more than just call
diff --git a/glib/glibmmconfig.h.in b/glib/glibmmconfig.h.in
index a36c293b..e5639cb7 100644
--- glib/glibmmconfig.h.in
+++ glib/glibmmconfig.h.in
@@ -22,6 +22,9 @@

 #ifdef GLIBMM_CONFIGURE

+/* Define only on Mac OS, COCOA */
+#undef GLIBMM_OS_COCOA
+
 /* Define if extern "C" and extern "C++" function pointers are compatible. */
 #undef GLIBMM_CAN_ASSIGN_NON_EXTERN_C_FUNCTIONS_TO_EXTERN_C_CALLBACKS

--
2.11.0
